### ISSUE TITLE
Can't deploy Smart contracts compiled by 0.8.20+ solidity (Shanghai Support PUSH0)

### ISSUE THREAD

**Issue by NickSneo** [hyperledger/besu#5468]
### Description
Smart contracts compiled with solidity 0.8.20+ with new OPCODE PUSH0 are failing to deploy on Besu 23.4.0 with error - `Error: Transaction has been reverted by the EVM` with default dev-genesis file

When I am using a modified dev-config genesis.json file to include "shanghaiTime" in my genesis file - 
```
{
  "config": {
    "chainId": 1337,
    "londonBlock": 0,
    "shanghaiTime": 10,
    "contractSizeLimit": 2147483647,
    "ethash": {
      "fixeddifficulty": 100
    }
  },
  "nonce": "0x42",
  "baseFeePerGas":"0x0",
  "timestamp": "0x0",
  "extraData": "0x11bbe8db4e347b4e8c937c1c8370e4b5ed33adb3db69cbdb7a38e1e50b1b82fa",
  "gasLimit": "0x1fffffffffffff",
  "difficulty": "0x10000",
  "mixHash": "0x0000000000000000000000000000000000000000000000000000000000000000",
  "coinbase": "0x0000000000000000000000000000000000000000",
  "alloc": {
    "fe3b557e8fb62b89f4916b721be55ceb828dbd73": {
      "privateKey": "8f2a55949038a9610f50fb23b5883af3b4ecb3c3bb792cbcefbd1542c692be63",
      "comment": "private key and this comment are ignored.  In a real chain, the private key should NOT be stored",
      "balance": "0xad78ebc5ac6200000"
    },
    "627306090abaB3A6e1400e9345bC60c78a8BEf57": {
      "privateKey": "c87509a1c067bbde78beb793e6fa76530b6382a4c0241e5e4a9ec0a0f44dc0d3",
      "comment": "private key and this comment are ignored.  In a real chain, the private key should NOT be stored",
      "balance": "90000000000000000000000"
    },
    "f17f52151EbEF6C7334FAD080c5704D77216b732": {
      "privateKey": "ae6ae8e5ccbfb04590405997ee2d52d2b330726137b875053c36d94e974d162f",
      "comment": "private key and this comment are ignored.  In a real chain, the private key should NOT be stored",
      "balance": "90000000000000000000000"
    }
  }
```
But this genesis.json gives me error - 
`2023-05-17 15:16:21.853+05:30 | pool-7-thread-1 | INFO  | MainnetBlockValidator | Invalid block 1 (0x029d319a0224571bc0c14dc13b5d735b2c2797772d6abf2cb01a7424d7d91e82): Optional[header validation rule violated, see logs]
2023-05-17 15:16:21.854+05:30 | pool-7-thread-1 | ERROR | BlockMiner | Illegal block mined, could not be imported to local chain.`

Now if I add `"terminalTotalDifficulty":0` in my genesis.json,
Blocks won't start to get mined, seems besu gets stuck

Smart Contract - 
```
// SPDX-License-Identifier: MIT
pragma solidity 0.8.20;

contract PushZeroContract {
    function pushZero() external pure returns (uint) {
        uint zero = 0;
        return zero;
    }
}
```
ABI = 
```
[
	{
		"inputs": [],
		"name": "pushZero",
		"outputs": [
			{
				"internalType": "uint256",
				"name": "",
				"type": "uint256"
			}
		],
		"stateMutability": "pure",
		"type": "function"
	}
]
```

Bytecode = "608060405234801561000f575f80fd5b5060b38061001c5f395ff3fe6080604052348015600e575f80fd5b50600436106026575f3560e01c8063e83b401e14602a575b5f80fd5b60306044565b604051603b91906066565b60405180910390f35b5f805f90508091505090565b5f819050919050565b6060816050565b82525050565b5f60208201905060775f8301846059565b9291505056fea2646970667358221220ca7b5654de4c4b61649abbf792e0dd121895f41f478845e0debf01b29be16e8764736f6c63430008140033"

Opcodes = "PUSH1 0x80 PUSH1 0x40 MSTORE CALLVALUE DUP1 ISZERO PUSH1 0xE JUMPI PUSH0 DUP1 REVERT JUMPDEST POP PUSH1 0x4 CALLDATASIZE LT PUSH1 0x26 JUMPI PUSH0 CALLDATALOAD PUSH1 0xE0 SHR DUP1 PUSH4 0xE83B401E EQ PUSH1 0x2A JUMPI JUMPDEST PUSH0 DUP1 REVERT JUMPDEST PUSH1 0x30 PUSH1 0x44 JUMP JUMPDEST PUSH1 0x40 MLOAD PUSH1 0x3B SWAP2 SWAP1 PUSH1 0x66 JUMP JUMPDEST PUSH1 0x40 MLOAD DUP1 SWAP2 SUB SWAP1 RETURN JUMPDEST PUSH0 DUP1 PUSH0 SWAP1 POP DUP1 SWAP2 POP POP SWAP1 JUMP JUMPDEST PUSH0 DUP2 SWAP1 POP SWAP2 SWAP1 POP JUMP JUMPDEST PUSH1 0x60 DUP2 PUSH1 0x50 JUMP JUMPDEST DUP3 MSTORE POP POP JUMP JUMPDEST PUSH0 PUSH1 0x20 DUP3 ADD SWAP1 POP PUSH1 0x77 PUSH0 DUP4 ADD DUP5 PUSH1 0x59 JUMP JUMPDEST SWAP3 SWAP2 POP POP JUMP INVALID LOG2 PUSH5 0x6970667358 0x22 SLT KECCAK256 0xCA PUSH28 0x5654DE4C4B61649ABBF792E0DD121895F41F478845E0DEBF01B29BE1 PUSH15 0x8764736F6C63430008140033000000 "

### Acceptance Criteria
* Should be able to deploy smart contracts compiled with solidity 0.8.20 with PUSH0 OPCODES

### Steps to Reproduce (Bug)
1. Start besu 23.4.0
2. Compile the given smart contract with solidity 0.8.20
3. Deploy the smart contract using any library or remix

**Expected behavior:** 
Should be able to deploy smart contracts compiled with solidity 0.8.20 with PUSH0 OPCODES

**Actual behavior:** 
Transaction fails with `Error: Transaction has been reverted by the EVM`

**Frequency:** 
always

### Versions (Add all that apply)
* Software version: besu/v23.4.0/osx-aarch_64/oracle_openjdk-java-19
* Java version: openjdk 19.0.2 2023-01-17
* OS Name & Version: MacOS 13.0 (22A380)
* Kernel Version: Darwin Nischals-MacBook-Pro.local 22.1.0 Darwin Kernel Version 22.1.0: Sun Oct  9 20:15:52 PDT 2022; root:xnu-8792.41.9~2/RELEASE_ARM64_T8112 arm64
* Virtual Machine software & version: [`vmware -v`]
* Docker Version: [`docker version`]
* Cloud VM, type, size: [Amazon Web Services I3-large]
* Consensus Client & Version if using Proof of Stake: [e.g. Teku, Lighthouse, Prysm, Nimbus, Lodestar]

### Smart contract information (If you're reporting an issue arising from deploying or calling a smart contract, please supply related information)
* Solidity version 0.8.20
* Repo with minimal set of deployable/reproducible contract code - please provide a link
* Please include specifics on how you are deploying/calling the contract 
* Have you reproduced the issue on other eth clients

### Additional Information (Add any of the following or anything else that may be relevant)
* Besu setup info - genesis file, config options
* System info - memory, CPU

**Comment by non-fungible-nelson** [comment]
@siladu - not sure where this bug lies, could be in the timestamp piece? if not maybe @shemnon has input.

**Comment by jframe** [comment]
Have been able to reproduce this locally as well. This is a different bug from #5446.

The cause of this is that we are building a POW block with nonce which and validating against Shanghai header validation rules which don't allow for nonce as Shanghai is expected to be POS. So not related to the fork timestamps that part is working.

We will need to make changes to Besu to support this.

**Comment by suraneti** [comment]
I faced this issue on Besu `22.10.3` with Solidity `0.8.20+`, and resolved it by reverting to version `0.8.19`.
